src/providers/authProvider/permissions.ts

```diff
@@ -28,9 +29,38 @@ const permissions: Record<UserRole, ResourcePermissions> = {
 
 export default permissions
 
```

Comment:
Hello @TahaKhanAbdalli - I think VAL can be simpler than this. I don't think VAL needs to think about roles or permissions. VAL can ignore these. I think we can remove the concept of permissions from VAL. We just make the request of Soul, and either Soul will return result, or return an error indicating "unauthorised access".

Aah, hold on.  The client-side does need to know about the current user permissions, since we need to know whether to provide create/edit buttons.  Hmm, we _could_ hard-code those in T/S (since that is how we did it previously).  Then, Soul-auth is a `fall-back` - only allowing REST interactions that the user is authorised to do.

Hmm, I think our clients would probably prefer this to be configured in SQLite, than hard-coded in T/S.

So, that's an argument for accessing roles and permissions in the way you've been doing.

Well done, Taha - carry on. 🤣 .

Any thoughts?

Reply:
Actually, we are rendering our UI according to `permissions`. So that's why we are doing this.